third_party/blink/renderer/modules/cobalt/h5vcc_metrics/h_5_vcc_metrics.idl

```diff
@@ -0,0 +1,61 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This interface mirrors the H5vccMetrics Web IDL defined in Cobalt LTS 25:
+// https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/25.lts.1+:cobalt/h5vcc/h5vcc_metrics.idl
+// Comments that are lifted verbatim from that implementation are quoted ("").
+[
+    Exposed=Window,
+    SecureContext
+]
+interface H5vccMetrics {
+  // "Interface for web client to bind an event handler that will be invoked
+  // every time Cobalt wants to upload a metrics payload. Only one event
+  // handler callback can be registered at any time. Duplicate calls to
+  // onMetricEvent will override any previously registered callback."
+  // This API changed to EventHandler. The original definition was:
+  //   void onMetricEvent(H5vccMetricEventHandler eventHandler);
+  attribute EventHandler onmetrics;
+
+  // "Enable Cobalt metrics logging. Metrics are disabled by default and should
+  // be enabled as soon as possible. When enabled, at regular intervals, the
+  // bound onMetricEvent callback will be called with telemetry payloads to
+  // be uploaded and logged on the server. Both the enable/disable settings
+  // are persistent or "sticky". That is, you only have to call them once
+  // and that setting will persist through multiple app lifecycles until the
+  // enable/disable APIs are explicitly called again."
+  [CallWith=ScriptState, RaisesException] Promise<void> enable();
+
+  // "Disable Cobalt metrics logging. If disabled, the metric event handler
+  // should never get called afterward."
+  [CallWith=ScriptState, RaisesException] Promise<void> disable();
+
+  // "Returns the current enabled state of metrics reporting. Note, the enable()
+  // and disable() APIs are asynchronous under the hood. This means if you call
+  // isEnabled() immediately after enable/disable, you may get unexpected
+  // results.
+  // TODO(b/290064552): Update this when H5vccMetrics APIs support async
+  // behavior."
+  // This API is now Promise-based.
+  [CallWith=ScriptState, RaisesException] Promise<boolean> isEnabled();
+
+  // "Sets the frequency in which Cobalt metrics will be snapshotted and sent to
+  // the metric event handler. Defaults to 5 minutes if not set. Must be > 0,
+  // else will be ignored. This setting is persistent. That is, it's "sticky"
+  // and will be persisted through multiple app lifetimes unless it's called
+  // again with a different value."
+  // This API is now Promise-based.
+  [ CallWith = ScriptState, RaisesException ] Promise<void>
```

Comment:
I'm not familiar with IDL style guidelines (if they even exist?) but it seems strange to have the return value on the same line as the decorators but a different line from the function it applies to

Reply:
This is my bad, I should have it in the next line (here and elsewhere). Maybe I copied directly the Cobalt25 code?

It made me wonder if we have a `git cl format` for IDL, hmm.

Reply:
Collapsed to a single (long) line like other examples around (in particular I looked at media_recorder.idl).